command/cmd_copy.go

```diff
@@ -147,6 +147,11 @@ func (c Copy) Run(ctx context.Context) error {
 	go func() {
 		defer close(errDoneCh)
 		for err := range waiter.Err() {
+			if strings.Contains(err.Error(), "too many open files") {
+				fmt.Println("WARNING You are hitting the max open file limit allowed by your OS. You can increase open file limit or try to decrease the number of workers with the parameter '-numworkers'.")
```

Comment:
Should we log these messages with log package?

```
msg := log.ErrorMessage{
	Err:       fmt.Errorf("error content here"),
}
log.Error(msg)
```
Also, custom message type and log level (WARNING in this case) can be added into log package if it is required.

Reply:
@sonmezonur on my tests that use log package, the tool exits without printing the logs. `logger.out()` is responsible for printing logs and it is running in a goroutine. 

Is there any way to drain the logs before exit?

Reply:
[log.Close()](https://github.com/peak/s5cmd/blob/master/command/app.go#L97) is responsible for closing the logger and draining messages.

Reply:
👍 done